frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -15,115 +15,158 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Implementation of `construct_runtime`.
+//!
+//! `construct_runtime` implementation is recursive and can generate code which will call itself in
+//! order to get all the pallet parts for each pallet.
+//!
+//! Pallets define their parts (`Call`, `Storage`, ..) either explicitly with the syntax
+//! `::{Call, ...}` or implicitly.
+//!
+//! In case a pallet define their part implicitly, then the pallet must provide the macro
+//! `construct_runtime_parts`. `construct_rutime` will generate some code which calls the macro
+//! `construct_runtime_parts` of the pallet. The `construct_runtime_parts` will generate some code
+//! which calls `construct_runtime` again with this time all explicit definition of parts.
+//!
+//! E.g.
+//! ```ignore
+//! construct_runtime!(
+//! 	...
+//! 	{
+//! 		System: frame_system = 0, // Implicit definition of parts
+//! 		Balances: pallet_balances = 1, // Implicit definition of parts
+//! 	}
+//! );
+//! ```
+//! This call has some implicit pallet parts, thus it will expand to:
+//! ```ignore
+//! pallet_balances::construct_runtime_parts!(
+//! 	// First *argument* is the path to frame_support crate.
+//! 	{ frame_support }
+//! 	// Second *argument* is the token idenfiying the pallet on which `construct_runtime_parts`
```

Comment:
```suggestion
//! 	// Second *argument* is the token identifying the pallet on which `construct_runtime_parts`
```

I'm also unsure what you mean here, do you mean that the 2nd argument should have pallet parts that needs to be included to the pallet?

Reply:
construct_runtime_parts is called like this:
```
construct_runtime_parts!( { some_token } { some_other_token } some_lasts_token );
```
at `some_other_token` (which I called the second *argument*) we need to pass where the macro should add the parts of the pallet. So we give some token identifying the place where we should add the parts.

The macro `construct_runtime_parts` will search for the tokens `some_other_token` inside `some_lasts_token`. Once it find a match. It will insert the parts `::{Call, Storage, ...}` and generate `some_lasts_token` with the parts inserted.

So in this example construct_runtime_parts search for the tokens `Balances: pallet_balances` inside 
```
frame_system::construct_runtime_parts!(
	{ frame_support }
	{ System: frame_system }
	construct_runtime!(
		...
		{
			System: frame_system = 0, // Implicit definition of parts
			Balances: pallet_balances = 1, // Implicit definition of parts
		}
	);
);
```
insert the part, so create the tokens:
```
frame_system::construct_runtime_parts!(
	{ frame_support }
	{ System: frame_system }
	construct_runtime!(
		...
		{
			System: frame_system = 0, // Implicit definition of parts
			Balances: pallet_balances::{Call, Storage, Pallet} = 1,
		}
	);
);
```
And expand to this new tokens.
The macro will output:
```
frame_system::construct_runtime_parts!(
	{ frame_support }
	{ System: frame_system }
	construct_runtime!(
		...
		{
			System: frame_system = 0, // Implicit definition of parts
			Balances: pallet_balances::{Call, Storage, Pallet} = 1,
		}
	);
);